app/agents_v2/primary_agent/reasoning/tool_intelligence.py

```
@@ -0,0 +1,468 @@
+            decision = ToolDecisionType.BOTH_SOURCES_NEEDED
+            reasoning = (
+                f"Both internal knowledge base and web search required. "
+                f"Identified {len(temporal_factors)} temporal factors and {len(knowledge_gaps)} knowledge gaps. "
+                f"Will combine internal expertise with current external information."
+            )
+        
+        elif needs_external_info:
+            if is_urgent:
+                decision = ToolDecisionType.BOTH_SOURCES_NEEDED
+                reasoning = (
+                    f"Urgent query requiring external information. "
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)